File: test_api.js
const http = require('http');

const url = 'http://192.168.137.1:5000/api/sonarr/series/recent';

console.log('🔍 Test de l\'API:', url);

http.get(url, (res) => {
  let data = '';
  console.log('📊 Status Code:', res.statusCode);
  res.on('data', (chunk) => {
    data += chunk;
  });
  res.on('end', () => {
    try {
      const json = JSON.parse(data);
      console.log('\n✅ Réponse JSON reçue:');
      console.log(JSON.stringify(json, null, 2));
      if (Array.isArray(json) && json.length > 0) {
        console.log(`\n📈 Nombre d\'éléments: ${json.length}`);
        console.log('🔍 Premier élément:');
        console.log(JSON.stringify(json[0], null, 2));
        console.log('\n🔑 Clés disponibles dans le premier élément:');
        Object.entries(json[0]).forEach(([key, value]) => {
          let type = Array.isArray(value) ? 'Array' : typeof value;
          let preview = '';
          if (typeof value === 'string' && value.length > 100) {
            preview = value.substring(0, 100) + '...';
          } else if (Array.isArray(value)) {
            preview = `Liste de ${value.length} éléments`;
          } else if (typeof value === 'object' && value !== null) {
            preview = `Objet avec ${Object.keys(value).length} clés`;
          } else {
            preview = value;
          }
          console.log(`  - ${key}: ${type} = ${preview}`);
        });
      }
    } catch (e) {
      console.error('❌ Erreur lors du parsing JSON:', e.message);
      console.log('📄 Contenu brut:', data);
    }
  });
}).on('error', (err) => {
  console.error('❌ Erreur de connexion:', err.message);
}); 

File: test_api.py
import requests
import json

def test_api():
    url = "http://192.168.137.1:5000/api/sonarr/series/recent"
    
    try:
        print("🔍 Test de l'API:", url)
        response = requests.get(url, timeout=10)
        
        print(f"📊 Status Code: {response.status_code}")
        print(f"📋 Headers: {dict(response.headers)}")
        
        if response.status_code == 200:
            data = response.json()
            print("\n✅ Réponse JSON reçue:")
            print(json.dumps(data, indent=2, ensure_ascii=False))
            
            # Analyser la structure
            if isinstance(data, list) and len(data) > 0:
                print(f"\n📈 Nombre d'éléments: {len(data)}")
                print(f"🔍 Premier élément:")
                print(json.dumps(data[0], indent=2, ensure_ascii=False))
                
                # Analyser les clés du premier élément
                if isinstance(data[0], dict):
                    print(f"\n🔑 Clés disponibles dans le premier élément:")
                    for key, value in data[0].items():
                        print(f"  - {key}: {type(value).__name__} = {value}")
        else:
            print(f"❌ Erreur HTTP: {response.status_code}")
            print(f"📄 Contenu de la réponse: {response.text}")
            
    except requests.exceptions.ConnectionError:
        print("❌ Erreur de connexion: Impossible de se connecter au serveur")
    except requests.exceptions.Timeout:
        print("⏰ Timeout: La requête a pris trop de temps")
    except json.JSONDecodeError as e:
        print(f"❌ Erreur JSON: {e}")
        print(f"📄 Contenu brut: {response.text}")
    except Exception as e:
        print(f"❌ Erreur inattendue: {e}")

if __name__ == "__main__":
    test_api() 